buildflow/core/resources/pulumi.py

```diff
@@ -1,18 +1,14 @@
 import dataclasses
-from typing import Any, Dict
 
 import pulumi
 
 from buildflow.core.resources._resource import Resource, ResourceID, ResourceType
 
 
-# NOTE: We currently dont do anything with this type, but we plan on using this
-# in the State API to track the Pulumi resources that are created.
 @dataclasses.dataclass
-class PulumiResource(Resource):
+class PulumiComponentResource(Resource):
```

Comment:
I wonder if we should just drop the Resource type completely now. Its not used for anything and is just another step to do for the user

Reply:
Maybe instead of making the ComponentResource classes in the Provider classes, we could define them all in the resources/ dir and import them in the provider classes?

Reply:
yea that's not a bad idea, than the providers really just contain the logic for creating the ComponentResources